packages/wallet/src/account.ts

```diff
@@ -86,11 +86,17 @@ export class Account extends AbstractAccount {
    * @param excludedIds - IDs of resources to be excluded from the query.
    * @returns A promise that resolves to an array of Resources.
    */
-  async getResourcesToSpend(
+  async getResourcesToSpend<T extends this>(
```

Comment:
This is a neat trick that lets the compiler infer which subclass is used to call the method. If it's `Predicate` calling it, the return would be `AccountResource<Predicate>[]`, and if it's `WalletUnlocked`, it would be `AccountResource<WalletUnlocked>[]`.